cuda/src/memory/cuda/resources/terminal/device_resource.cpp

```diff
@@ -0,0 +1,36 @@
+#include <memory>
+
+#include <cuda_runtime_api.h>
```

Comment:
Principle of least action, if `cuda_runtime_api.h` has everything we need (which I believe it does) then I would rather import that than the heavier `cuda_runtime.h`, since both are designed to be outward-facing headers for user code. However for the sake of consistency I am more than happy to use `cuda_runtime.h`! :slightly_smiling_face: